### PR TITLE
Add VectorN.Random functions

### PR DESCRIPTION
**Pull request by dr-vortex**
Adds functions for Vector2, Vector3, and Vector4 to get a vector with random values

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13532/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13532/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13532/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13532/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13532/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13532/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13532/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13532/merge#BCU1XR#0

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/Maths/math.vector.ts [hunk 1]
```diff
@@ -494,6 +494,14 @@ export class Vector2 {
         return new Vector2(1, 1);
     }
 
+	/**
+	 * Returns a new Vector2 with random values
```
Comment (Popov72) [inline packages/dev/core/src/Maths/math.vector.ts:498]: I think the doc should explicitely state that the random values are between 0 and 1.

**Review by RaananW — APPROVED** [review]
LGTM!
I find the comment correct - we should either define that random values are between 0 and 1 OR allow providing min/max values. If min/max, add a private "_getRandom(min, max)" function that roughly looks like this:

`return min + Math.random() * (max - min);`

or this - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#getting_a_random_number_between_two_values)

Shame that js random doesn't have a seed for the random function...

File: packages/dev/core/src/Maths/math.vector.ts (hunk 1 above)
Comment (dr-vortex) [reply packages/dev/core/src/Maths/math.vector.ts]: Updated - Now accepts (and documents) a min/max

**Comment by dr-vortex** [comment]
@RaananW 
No need for `_getRandom` since we have `Scalar.RandomRange`.
I've updated the code to accept a min and max (which default to 0 and 1).

**Review by sebavan — APPROVED** [review]

**Comment by jelster** [comment]
I came too late to this party, but I am curious if there's been any discussion about the need to support other RNG's than JS's `Math.random()`? For example, in Space-Truckers, I implemented [a gaussian distribution RNG](https://github.com/jelster/space-truckers/blob/develop/src/randomGenerator.js) - would that be something that would make sense to add to `Scalar.RandomXXX` functions an optional function parameter that accepts a distribution function to use?

**Comment by dr-vortex** [comment]
@jelster If you want you can open a PR to add that. Maybe an optional 3d parameter as a RNG function?

**Comment by dr-vortex** [comment]
@jelster If you want you can open a PR to add that. Maybe an optional 3d parameter as a RNG function?

The only issue is the min/max parameters (since a RNG function *must* implement those as a first and second parameter.

**Comment by jelster** [comment]
Yah, I can work on that. 👍 

What, if any, RNG functions would we want to provide OOB other than the base and Gaussian?

Another question - should we attempt to memoize the RNG function passed in? I feel like it could be premature optimization, and that we would want developers to be able to use different functions in different places of their application. Still, since it is a function that can potentially be called 1000's of times, I wanted to raise the topic just in case 😄

**Comment by dr-vortex** [comment]
@jelster Great idea! If you want to open a PR I would recommend the below:
```ts
type RandomNumberGenerator = (min: number, max: number, ...options: number[]) => number;
```

**Comment by RaananW** [comment]
> Yah, I can work on that. 👍
> 
> What, if any, RNG functions would we want to provide OOB other than the base and Gaussian?
> 
> Another question - should we attempt to memoize the RNG function passed in? I feel like it could be premature optimization, and that we would want developers to be able to use different functions in different places of their application. Still, since it is a function that can potentially be called 1000's of times, I wanted to raise the topic just in case 😄

just to be sure - we don't want to add a new RNG to the babylon codebase, but to allow injecting an external one, if provided. it should take a function that might have optional parameters (min, max) and returns a number.

**Comment by dr-vortex** [comment]
That's what I was thinking as well.

**Comment by jelster** [comment]
> we don't want to add a new RNG to the babylon codebase, but to allow injecting an external one, if provided. it should take a function that might have optional parameters (min, max) and returns a number.

Yeah that's my understanding. I was thinking of providing the Gaussian function as an example RNG though - where should that go?

**Comment by dr-vortex** [comment]
You would add it as a third parameter. For example:
```ts
Vector2.Random(min: number, max: number, rng: (min: number, max: number) => number = Scalar.RandomRange){
	return new Vector2(rng(min, max), rng(min, max));
}
```

**Comment by RaananW** [comment]
Let's discuss that in the forum instead of here in an open issue.
This is not the way I would see it injected

**Pull request merged**